src/solvers/flattening/flatten_byte_operators.cpp

```diff
@@ -549,10 +549,12 @@ exprt flatten_byte_update(
     // zero-extend the value, but only if needed
     exprt value_extended;
 
-    if(width>integer2unsigned(element_size)*8)
+    if(width > element_size * 8)
```

Comment:
This looks inconsistent: `width` is a `size_t` (which is parsed with `integer2size_t` above). `element_size` is an `mp_integer`.

Reply:
Thus everything will be converted to `mp_integer`, which avoids falling for an (unsigned) integer overflow.